### ISSUE TITLE
Formula error in [terrain_defaults] crashes game

### ISSUE THREAD

**Issue by ProditorMagnus** [wesnoth/wesnoth#9672]
### Game and System Information

- **Version:** 1.19.7
- **OS:** windows

### Description of the bug

```
[units]
	[terrain_defaults]
		id=walkable_lava
		[movement_costs]
			default="( (if(debug_float(loc(44,23),furnacefoot) = null(), unwalkable, min(unwalkable,furnacefoot))) )" # 1
			# default="( (if(debug(loc(44,23),furnacefoot) = null(), unwalkable, min(unwalkable,furnacefoot))) )" # 2
		[/movement_costs]
		[defense]
			default="( if(furnacefoot = null(), unwalkable, min(unwalkable,furnacefoot)) )"
		[/defense]
	[/terrain_defaults]
[/units]
```

### Steps to reproduce the behavior

Test case 1 with debug_float crashes game without error message, probably because map is not available yet.
Test case 2 with debug crashes game with error message about invalid formula.

### Expected behavior

Affected addon should be disabled, rest of game should not be affected. Game should not crash.

**Comment by CelticMinstrel** [comment]
> probably because map is not available yet.

Why would it require the map? I don't see anything in that formula that depends on the map.

**Comment by ProditorMagnus** [comment]
I think `debug_float(loc(44,23)` needs map.

**Comment by gfgtdf** [comment]
> Affected addon should be disabled

this will probably be hard to implement, most likeley a fix will just print an error and return 0 in the forunla.

**Comment by CelticMinstrel** [comment]
Yeah, if `debug_float` can't function without a map, it should just return 0 or something when called without a map.
